tapgarden/custodian.go

```diff
@@ -698,6 +733,19 @@ func (c *Custodian) RegisterSubscriber(receiver *fn.EventReceiver[fn.Event],
 	return nil
 }
 
+// publishSubscriberStatusEvent publishes an event to all status events
+// subscribers.
+func (c *Custodian) publishSubscriberStatusEvent(event fn.Event) {
+	// Lock the subscriber mutex to ensure that we don't modify the
+	// subscriber map while we're iterating over it.
+	c.statusEventsSubsMtx.Lock()
+	defer c.statusEventsSubsMtx.Unlock()
+
+	for _, sub := range c.statusEventsSubs {
+		sub.NewItemCreated.ChanIn() <- event
```

Comment:
is this non-blocking?

if it's blocking then maybe worth considering bringing down the ctx to this function to ensure healthy exit

Reply:
Good point! I'll use `fn.SendOrQuit`.

Reply:
Looks like we'd want to use `SendOrQuit` for all event publishing, not just here? Could be follow-up PR.

Reply:
> SendOrQuit

Good idea!